ledger-core/network/messagesender/service.go

```diff
@@ -42,12 +44,18 @@ type Service interface {
 	// blocks if network unreachable
 	SendRole(ctx context.Context, msg rmsreg.GoGoSerializable, role affinity.DynamicRole, object reference.Global, pn pulse.Number, opts ...SendOption) error
 	SendTarget(ctx context.Context, msg rmsreg.GoGoSerializable, target reference.Global, opts ...SendOption) error
+
+	InterceptorAdd(fn InterceptorFn)
+	InterceptorClear()
 }
 
 type DefaultService struct {
 	pub      message.Publisher
 	affinity affinity.Helper
 	pulses   beat.History
+
+	interceptorsLock sync.RWMutex
```

Comment:
imho - it could be better to have 2 implementations - with and without interception

Reply:
fixed